### PR TITLE
Health reporting for Emby

### PR DESCRIPTION
**Pull request by softworkz**

### CODE REVIEW

File: MediaBrowser.Controller/Health/HealthMessage.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        }
+
+        /// <summary>
+        /// Gets the type of health message.
+        /// </summary>
+        public string MessageType
+        {
+            get
+            {
+                return _messageType;
```
Comment (drewfreyling) [inline MediaBrowser.Controller/Health/HealthMessage.cs:72]: Why don't you use auto properties here?

Comment (softworkz) [inline MediaBrowser.Controller/Health/HealthMessage.cs:72]: Because those values are immutable and auto-properties are unable to express this.

Comment (drewfreyling) [inline MediaBrowser.Controller/Health/HealthMessage.cs:72]: You can get the same value out of `public string MessageType {get; private set;}`

Comment (softworkz) [inline MediaBrowser.Controller/Health/HealthMessage.cs:72]: No. This restricts access to the setter only, but doesn't make the values immutable.

Comment (drewfreyling) [inline MediaBrowser.Controller/Health/HealthMessage.cs:72]: Ah yes sorry you are right. I guess we could use expression bodies here if the CI supports C# 6 - `public string MessageType  => _messageType;`

Comment (softworkz) [inline MediaBrowser.Controller/Health/HealthMessage.cs:72]: Immutability is not a functional requirement. Declaring those fields as readonly may affect the compilation but probably with little to none benefit regarding performance.
The primary reason for declaring those fields as readonly is to improve clarity of code and make it easy to understand.  We could find a number of fancy alternatives, but none of them would meet this requirement any better...

**Comment by softworkz** [comment]
Adjusted to removal of paper-fab

**Comment by drewfreyling** [comment]
What is happening with this PR? @softworkz it has merge conflicts.

**Pull request closed without merging**